
<!DOCTYPE html>
<html lang='en'>
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
    <title>Tree - rpms/ganglia - src.fedoraproject.org</title>
        <link rel="shortcut icon" type="image/vnd.microsoft.icon"
        href="/theme/static/favicon.ico?version=5.13.3"/>
    <link href="/theme/static/fedora-bootstrap-1.3.0/fedora-bootstrap.min.css?version=5.13.3"
        type="text/css" rel="stylesheet" />
    <link href="/theme/static/fonts/fonts.css?version=5.13.3"
        rel="stylesheet" type="text/css" />
    <link href="/theme/static/fonts/hack_fonts/css/hack-extended.min.css?version=5.13.3"
        type="text/css" rel="stylesheet" />
    <link href="/theme/static/theme.css?version=5.13.3"
        type="text/css" rel="stylesheet" />

    <link type="text/css" rel="stylesheet" nonce="x8dmwigkFAnlN60QkSaKTGdHp" href="/static/vendor/font-awesome/font-awesome.css?version=5.13.3"/>
    <link type="text/css" rel="stylesheet" nonce="x8dmwigkFAnlN60QkSaKTGdHp" href="/static/pagure.css?version=5.13.3"/>
<link nonce="x8dmwigkFAnlN60QkSaKTGdHp" rel="stylesheet" href="/static/vendor/highlight.js/styles/github.css?version=5.13.3"/>
<link nonce="x8dmwigkFAnlN60QkSaKTGdHp" rel="stylesheet" href="/static/vendor/highlightjs-line-numbers/highlightjs-line-numbers.min.css?version=5.13.3"/>
<style nonce="x8dmwigkFAnlN60QkSaKTGdHp">
  .hljs {
    background: #fff;
  }
</style>
  </head>
  <body id="home">
    
    <!-- start masthead -->
    <nav class="navbar navbar-light masthead p-0 navbar-expand">
      <div class="container">
        <a href="/" class="navbar-brand">
        <img height="40" src="/theme/static/pagure-logo.png?version=5.13.3"
             alt="pagure Logo" id="pagureLogo"/>
        </a>
        <ul class="navbar-nav ml-auto">

          

          <li class="nav-item">
            <a class="btn btn-primary" href="/login/?next=https://src.fedoraproject.org/rpms/ganglia/blob/f32/f/ganglia-web-5ee6b7.patch">Log In</a>
          </li>
        </ul>
      </div>
    </nav>
    <!-- close masthead-->

    <div class="bodycontent">


<div class="bg-light border border-bottom pt-3">
  <div class="container">
    <div class="row mb-3">
      <div class="col-6">
        <div class="row">
        <div class="col-auto pr-0">
            <h3>
<i class="fa fa-archive text-muted"></i></h3>
        </div>
        <div class="col-auto pl-2">
            <h3 class="mb-0">
<a href="/projects/rpms/%2A">rpms</a>&nbsp;/&nbsp;<a href="/rpms/ganglia"><strong>ganglia</strong></a>
            </h3>
        </div>
        </div>
      </div>
      <div class="col-6 text-right">
        <div class="btn-group">
        <div class="btn-group">
        <a href="#"
            class="btn btn-sm dropdown-toggle btn-outline-primary"
            data-toggle="dropdown" id="watch-button">
          <i class="fa fa-clone fa-fw"></i>
          <span>Clone</span>
        </a>
        <div class="dropdown-menu dropdown-menu-right">
          <div class="m-3" id="source-dropdown">
            <div>
              <h5><strong>Source Code</strong></h5>

              <div class="form-group">
                <div class="input-group input-group-sm">
                  <div class="input-group-prepend"><span class="input-group-text">GIT</span></div>
                  <input class="form-control bg-white select-on-focus" type="text" value="https://src.fedoraproject.org/rpms/ganglia.git" readonly>
                </div>
              </div>
            </div>
          </div>
        </div>

      </div>
    </div>
  </div>
</div>

<ul class="nav nav-tabs nav-small border-bottom-0">
  <li class="nav-item mr-2 text-dark">
    <a class="nav-link active" href="/rpms/ganglia">
        <i class="fa fa-code fa-fw text-muted"></i>
        <span class="d-none d-md-inline">Source</span>
    </a>
  </li>

    <li class="nav-item mr-2 text-dark">
        <a class="nav-link" href="https://bugzilla.redhat.com/buglist.cgi?bug_status=NEW&bug_status=ASSIGNED&classification=Fedora&product=Fedora&product=Fedora EPEL&component=ganglia">
          <i class="fa fa-fw text-muted fa-exclamation-circle"></i>
          <span class="d-none d-md-inline">Issues&nbsp;</span>
          <span class="fa fa-external-link"></span>
        </a>
    </li>

    <li class="nav-item mr-2 text-dark">
      <a class="nav-link" href="/rpms/ganglia/pull-requests">
          <i class="fa fa-fw text-muted fa-arrow-circle-down"></i>
          <span class="d-none d-md-inline">Pull Requests&nbsp;</span>
          <span class="badge badge-secondary py-0 d-none d-md-inline">
            0
          </span>
      </a>
    </li>


    <li class="nav-item mr-2 text-dark">
      <a class="nav-link" href="/rpms/ganglia/stats">
          <i class="fa fa-line-chart fa-fw text-muted"></i>
          <span class="d-none d-md-inline">Stats</span>
      </a>
    </li>


</ul>
  </div>
</div>

<div class="container pt-5 repo-body-container">
  <div class="row">
    <div class="col-2">
<nav class="nav nav-tabs nav-sidetabs flex-column">
  <a class=
      "nav-link nowrap
"
      href="/rpms/ganglia">
      <i class="fa fa-home text-muted fa-fw"></i>&nbsp;<span class="d-none d-md-inline">Overview</span>
  </a>
  <a class=
    "nav-link nowrap
 active"
    href="/rpms/ganglia/tree/f32">
    <i class="fa fa-file-code-o text-muted fa-fw"></i>&nbsp;Files
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/ganglia/commits/f32">
    <i class="fa fa-list-alt text-muted fa-fw" data-glyph="spreadsheet"></i>&nbsp;Commits
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/ganglia/branches?branchname=f32">
    <i class="fa fa-random text-muted fa-fw"></i>&nbsp;Branches
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/ganglia/forks">
    <i class="fa fa-code-fork text-muted fa-fw"></i>&nbsp;Forks
  </a>
  <a class=
    "nav-link nowrap
"
    href="/rpms/ganglia/releases">
    <i class="fa fa-tags text-muted fa-fw"></i>&nbsp;Releases
  </a>

  <div class="col-xs-2 line-height-1"></div>
  <h6>Monitoring status:</h6>
  <div class="btn-group">
    <button title="Monitoring status" class="btn btn-sm btn-outline-primary disabled"
        id="monitoring-button">
      <i id="monitoring-icon" class="fa fa-fw fa-eye"></i>
      <span id="monitoring-label" class="fa fa-circle-o-notch fa-spin fa-1x fa-fw"></span>
    </button>
  </div>

  <div class="col-xs-2 line-height-1"></div>
  <div id="orphan-section" class="pt-3">
      <div class="col-xs-2 line-height-1"></div>

  </div>

  <div class="pt-3">
    <div class="col-xs-2 line-height-1">
    <h6>Bugzilla Assignee:</h6>
      <dl>
        <dt>Fedora: </dt>
        <dd id="fedora_assignee_txt">
        terjeros
        </dd>
        <dt>EPEL: </dt>
        <dd id="epel_assignee_txt">
        noodles
        </dd>
      </dl>
    </div>
  </div>

  <div class="modal fade" id="modal_assignee" tabindex="-1"
          role="dialog" aria-labelledby="Bugzilla assignee" aria-hidden="true">
    <div class="modal-dialog" role="document">
      <div class="modal-content">
        <div class="modal-header">
          <h4 class="modal-title">Bugzilla Assignee</h4>
          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
            <span class="sr-only">Close</span>
          </button>
        </div>
        <form id="change_assignees">
          <div class="modal-body">
            <label for="fedora_assignee">Fedora</label>
            <input title="Default assignee for Fedora in bugzilla - Empty input resets to default"
              class="form-control" name="fedora_assignee" id="fedora_assignee" value="terjeros"/>
            <label for="epel_assignee">EPEL</label>
            <input title="Default assignee for EPEL in bugzilla (if applicable) - Empty input resets to default"
              class="form-control" name="epel_assignee" id="epel_assignee" value="noodles" />
            <p class="pt-2">
              These two fields allow to specify a different default assignee for ticket opened against
              this package in bugzilla. Note: The EPEL field is
              always displayed for packages in the 'rpms' namespace regardless of whether it
              is used in bugzilla or not.            </p>
          </div>
          <div class="modal-footer">
            <button class="btn btn-secondary" type="button" title="Update bugzilla overrides" id="reset_assignees">
               Reset to defaults
            </button>
            <button class="btn btn-primary" type="submit" title="Update bugzilla overrides" id="update_assignees">
               Update
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>

  <div class="modal fade" id="modal_orphan" tabindex="-1"
          role="dialog" aria-labelledby="Orphan this package" aria-hidden="true">
    <div class="modal-dialog" role="document">
      <div class="modal-content">
        <div class="modal-header">
          <h4 class="modal-title">Orphan package</h4>
          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
            <span class="sr-only">Close</span>
          </button>
        </div>
          <div class="modal-body">
            <div class="form-group">
              <label for="reason_menu">Reason</label>
              <select id="reason_menu" class="form-control">

                <option id="lack_of_time_option_button">Lack of time</option>

                <option id="do_not_use_it_option_button">Do not use it anymore</option>

                <option id="unmaintained_option_button">Unmaintained upstream</option>

                <option id="fails_to_build_option_button">Fails to build from source</option>

                <option id="not_fixed_option_button">Important bug not fixed</option>

                <option id="other_option_button">Other</option>
              </select>
            </div>
          <label for="orphan_reason_info">Additional info</label>
          <input title="Additional info for orphaning reason"
                  class="form-control" name="orphan_reason_info" id="orphan_reason_info" value=""/>
          </div>
          <div class="modal-footer">
            <button class="btn btn-primary" type="submit" title="Orphan this package" id="orphan_button">
               Update
            </button>
          </div>
      </div>
    </div>
  </div>

  <script type="text/javascript" nonce="x8dmwigkFAnlN60QkSaKTGdHp">
    window.addEventListener('load', function() {
      set_up_monitoring = function(status){
        var _label = "Disabled"
        if (status === "monitoring") {
          _label = "Monitoring";
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else if (status === "monitoring-with-scratch") {
          _label = "Scratch builds"
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye")
        } else {
          $("#monitoring-icon").attr("class", "fa fa-fw fa-eye-slash")
        }

        $("#monitoring-label").text(_label);
        $("#monitoring-label").removeClass("fa fa-circle-o-notch fa-spin fa-1x fa-fw");
      }

      $.ajax({
        url: "/_dg/anitya/rpms/ganglia",
        type: 'GET',
        dataType: 'json',
        success: function(res) {
          console.log(res);
          set_up_monitoring(res.monitoring)
        }
      });


      $("#reset_assignees").on('click', function(){
        $('#fedora_assignee').val('');
        $('#epel_assignee').val('');
        $("#change_assignees").submit();
        return false;
      });

      $("#change_assignees").on('submit',  function(){
        $('html').css('cursor', 'progress');
        $('#reset_assignees').attr('disabled', true);
        $('#update_assignees').attr('disabled', true);
        $('#update_assignees').text('Updating...');
        $.ajax({
          url: "/_dg/bzoverrides/rpms/ganglia",
          type: 'POST',
          dataType: 'json',
          data: {
             'epel_assignee': $('#epel_assignee').val(),
             'fedora_assignee': $('#fedora_assignee').val()
          },
          success: function(res) {
            $("#fedora_assignee_txt").text(res.fedora_assignee);
            $("#epel_assignee_txt").text(res.epel_assignee);
            $('#modal_assignee').modal('hide');
            $('#reset_assignees').attr('disabled', false);
            $('#update_assignees').attr('disabled', false);
            $('#update_assignees').text('Update');
            $('html').css('cursor', 'default');
            console.log("Successfully changed the bugzilla assignees");
            return false;
          },
          error: function(res) {
            var msg = '';
            if(res.responseJSON.errors){
              msg = ': ' + res.responseJSON.errors.join(', ');
            }
            alert("Unable to update the bugzilla assignee(s)" + msg);
            $('html').css('cursor', 'default');
            $('#reset_assignees').attr('disabled', false);
            $('#update_assignees').attr('disabled', false);
            $('#update_assignees').text('Update');
            return false;
          }
        })
        return false;
      });


      $.ajax({
        url: "/_dg/actived/rpms/ganglia",
        type: 'GET',
        dataType: 'json',
        success: function(res) {
          var _btn = $("#take_orphan_button");
          if (!res.active){
            _btn.off("click");
            _btn.click(function(){
                window.open(
                    "https://pagure.io/releng/new_issue?title="
                    + "Unretire rpms/ganglia"
                    + "&amp;template=package_unretiremet");
            });
            _btn.prop( "title", "Package retired - Open a releng ticket to adopt it" );
            _btn.html("Retired");
          }
          _btn.removeClass('disabled');
        }
      });

      $("#orphan_button").click(function(){
        $("#orphan_button").attr("disabled", true);
        $.ajax({
          url: "/_dg/orphan/rpms/ganglia",
          type: 'POST',
          dataType: 'json',
          data: {
            'orphan_reason': $('#reason_menu').val(),
            'orphan_reason_info': $('#orphan_reason_info').val()
          },
          success: function(res) {
            $("#point_of_contact_div").html("Package is currently unmaintained");
            $("#orphan_button").attr("disabled", false);
            $('#modal_orphan').modal('hide');
            $('#orphan-section').html('');
          },
          error: function(res) {
            if (res.responseJSON.errors) {
              alert('Unable to orphan the package: ' + res.responseJSON.errors);
            } else {
              alert('Unable to orphan the package: ' + res.responseJSON.error);
            }
            $("#orphan_button").attr("disabled", false);
          }
        });
      });
    });
  </script>

</nav>    </div>
    <div class="col-10">
  <div class="row mb-1">
    <div class="col-sm-6">
    <h3>
      Files
    </h3>
    </div>

    <div class="col-sm-6">
      <div class="float-right">
          <div class="btn-group">
            <a href="#" class="btn btn-outline-light border-secondary text-dark btn-sm dropdown-toggle"
                    data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                    <span class="fa fa-random fa-fw"></span> Branch: <span class="font-weight-bold">f32</span>
          </a>
            <div class="dropdown-menu dropdown-menu-right">
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/el4">
                      <span class="fa fa-random fa-fw"></span> <span class="">el4</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/el5">
                      <span class="fa fa-random fa-fw"></span> <span class="">el5</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/el6">
                      <span class="fa fa-random fa-fw"></span> <span class="">el6</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/epel7">
                      <span class="fa fa-random fa-fw"></span> <span class="">epel7</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/epel8">
                      <span class="fa fa-random fa-fw"></span> <span class="">epel8</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/epel8-playground">
                      <span class="fa fa-random fa-fw"></span> <span class="">epel8-playground</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/epel9">
                      <span class="fa fa-random fa-fw"></span> <span class="">epel9</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f10">
                      <span class="fa fa-random fa-fw"></span> <span class="">f10</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f11">
                      <span class="fa fa-random fa-fw"></span> <span class="">f11</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f12">
                      <span class="fa fa-random fa-fw"></span> <span class="">f12</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f13">
                      <span class="fa fa-random fa-fw"></span> <span class="">f13</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f14">
                      <span class="fa fa-random fa-fw"></span> <span class="">f14</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f15">
                      <span class="fa fa-random fa-fw"></span> <span class="">f15</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f16">
                      <span class="fa fa-random fa-fw"></span> <span class="">f16</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f17">
                      <span class="fa fa-random fa-fw"></span> <span class="">f17</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f18">
                      <span class="fa fa-random fa-fw"></span> <span class="">f18</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f19">
                      <span class="fa fa-random fa-fw"></span> <span class="">f19</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f20">
                      <span class="fa fa-random fa-fw"></span> <span class="">f20</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f21">
                      <span class="fa fa-random fa-fw"></span> <span class="">f21</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f22">
                      <span class="fa fa-random fa-fw"></span> <span class="">f22</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f23">
                      <span class="fa fa-random fa-fw"></span> <span class="">f23</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f24">
                      <span class="fa fa-random fa-fw"></span> <span class="">f24</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f25">
                      <span class="fa fa-random fa-fw"></span> <span class="">f25</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f26">
                      <span class="fa fa-random fa-fw"></span> <span class="">f26</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f27">
                      <span class="fa fa-random fa-fw"></span> <span class="">f27</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f28">
                      <span class="fa fa-random fa-fw"></span> <span class="">f28</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f29">
                      <span class="fa fa-random fa-fw"></span> <span class="">f29</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f30">
                      <span class="fa fa-random fa-fw"></span> <span class="">f30</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f31">
                      <span class="fa fa-random fa-fw"></span> <span class="">f31</span>
                  </a>
                  <a class="dropdown-item pl-1 active" href="/rpms/ganglia/tree/f32">
                      <span class="fa fa-random fa-fw"></span> <span class="font-weight-bold">f32</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f33">
                      <span class="fa fa-random fa-fw"></span> <span class="">f33</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f34">
                      <span class="fa fa-random fa-fw"></span> <span class="">f34</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f35">
                      <span class="fa fa-random fa-fw"></span> <span class="">f35</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f36">
                      <span class="fa fa-random fa-fw"></span> <span class="">f36</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f37">
                      <span class="fa fa-random fa-fw"></span> <span class="">f37</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f38">
                      <span class="fa fa-random fa-fw"></span> <span class="">f38</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f7">
                      <span class="fa fa-random fa-fw"></span> <span class="">f7</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f8">
                      <span class="fa fa-random fa-fw"></span> <span class="">f8</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/f9">
                      <span class="fa fa-random fa-fw"></span> <span class="">f9</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/fc6">
                      <span class="fa fa-random fa-fw"></span> <span class="">fc6</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/main">
                      <span class="fa fa-random fa-fw"></span> <span class="">main</span>
                  </a>
                  <a class="dropdown-item pl-1 " href="/rpms/ganglia/tree/rawhide">
                      <span class="fa fa-random fa-fw"></span> <span class="">rawhide</span>
                  </a>
            </div>
          </div>
    </div>
  </div>
  </div>
    <div class="card mb-3">
      <div class="card-header">
        <ol class="breadcrumb p-0 bg-transparent mb-0">
          <li class="breadcrumb-item">
            <a href="/rpms/ganglia/tree/f32">
              <span class="fa fa-random">
              </span>&nbsp; f32
            </a>
          </li>
          <li class="active breadcrumb-item">
            <span class="fa fa-file" data-glyph="">
            </span>&nbsp; ganglia-web-5ee6b7.patch
          </li>
        </ol>
      </div>

  <div class="card-body p-0">
            <div class="bg-light border text-right pr-2">
                <form class="btn btn-sm" method="POST" name="fork_project"
                    action="/fork_edit/rpms/ganglia/edit/f32/f/ganglia-web-5ee6b7.patch">
                    <button class="btn btn-sm btn-secondary fork_project_btn">
                            Fork and Edit
                    </button>
                    
                </form>

                <a class="btn btn-secondary btn-sm" href="/rpms/ganglia/blob/f32/f/ganglia-web-5ee6b7.patch" title="View as blob">Blob</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/ganglia/blame/ganglia-web-5ee6b7.patch?identifier=f32" title="View git blame">Blame</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/ganglia/history/ganglia-web-5ee6b7.patch?identifier=f32" title="View git log for this file">History</a>

                <a class="btn btn-secondary btn-sm" href="/rpms/ganglia/raw/f32/f/ganglia-web-5ee6b7.patch" title="View as raw">Raw</a>
            </div>

        <pre class="syntaxhighlightblock"><code class="lang-diff">diff --git a/conf_default.php.in b/conf_default.php.in
index 9efa9ef..f7602cd 100644
--- a/conf_default.php.in
+++ b/conf_default.php.in
@@ -135,6 +135,7 @@ $conf[&#39;cpu_gnice_color&#39;] = &#34;fff261&#34;;
 # Colors for the MEMORY report graph
 #
 $conf[&#39;mem_used_color&#39;] = &#34;5555cc&#34;;
+$conf[&#39;mem_available_color&#39;] = &#34;2030F4&#34;;
 $conf[&#39;mem_shared_color&#39;] = &#34;0000aa&#34;;
 $conf[&#39;mem_cached_color&#39;] = &#34;33cc33&#34;;
 $conf[&#39;mem_slab_color&#39;] = &#34;66e533&#34;;
diff --git a/functions.php b/functions.php
index be9fa4a..afbb586 100644
--- a/functions.php
+++ b/functions.php
@@ -1121,8 +1121,8 @@ function build_aggregate_graph_config ($graph_type,
                                        $line_width,
                                        $hreg,
                                        $mreg,
-                                       $glegend,
-                                       $exclude_host_from_legend_label,
+                                       $glegend = NULL,
+                                       $exclude_host_from_legend_label = false,
                                        $sortit = true) {
 
   global $conf, $index_array, $hosts, $grid, $clusters, $debug, $metrics;
diff --git a/graph.d/mem_report.php b/graph.d/mem_report.php
index d443965..58d167d 100644
--- a/graph.d/mem_report.php
+++ b/graph.d/mem_report.php
@@ -48,6 +48,8 @@ function graph_mem_report ( &amp;$rrdtool_graph ) {
     $bmem_slab_defs = &#39;&#39;;
     $bmem_buffers_defs = &#39;&#39;;
     $bmem_used_cdef = &#34;CDEF:&#39;bmem_used&#39;=&#39;bmem_total&#39;,&#39;bmem_free&#39;,-,&#39;bmem_cached&#39;,-&#34;;
+    $bmem_available_defs = &#39;&#39;;
+    $bmem_realused_cdef = &#39;&#39;;
 
     if (file_exists(&#34;$rrd_dir/mem_shared.rrd&#34;)) {
        $bmem_used_cdef .= &#34;,&#39;bmem_shared&#39;,UN,0,&#39;bmem_shared&#39;,IF,-&#34;;
@@ -67,6 +69,12 @@ function graph_mem_report ( &amp;$rrdtool_graph ) {
            .&#34;CDEF:&#39;bmem_buffers&#39;=mem_buffers,1024,* &#34;;
     }
 
+    if (file_exists(&#34;$rrd_dir/mem_available.rrd&#34;)) {
+       $bmem_realused_cdef = &#34;CDEF:&#39;bmem_realused&#39;=&#39;bmem_total&#39;,&#39;bmem_available&#39;,- &#34;;
+       $bmem_available_defs = &#34;DEF:&#39;mem_available&#39;=&#39;${rrd_dir}/mem_available.rrd&#39;:&#39;sum&#39;:AVERAGE &#34;
+           .&#34;CDEF:&#39;bmem_available&#39;=mem_available,1024,* &#34;;
+    }
+
     $series = &#34;DEF:&#39;mem_total&#39;=&#39;${rrd_dir}/mem_total.rrd&#39;:&#39;sum&#39;:AVERAGE &#34;
         .&#34;CDEF:&#39;bmem_total&#39;=mem_total,1024,* &#34;
         .$bmem_shared_defs
@@ -76,6 +84,8 @@ function graph_mem_report ( &amp;$rrdtool_graph ) {
         .&#34;CDEF:&#39;bmem_cached&#39;=mem_cached,1024,* &#34;
         .$bmem_slab_defs
         .$bmem_buffers_defs
+        .$bmem_available_defs
+        .$bmem_realused_cdef
         .&#34;$bmem_used_cdef &#34;
         .&#34;AREA:&#39;bmem_used&#39;#${conf[&#39;mem_used_color&#39;]}:&#39;Use${rmspace}&#39; &#34;;
 
@@ -188,6 +198,22 @@ function graph_mem_report ( &amp;$rrdtool_graph ) {
 	}
     }
 
+    if (file_exists(&#34;$rrd_dir/mem_available.rrd&#34;)) {
+        $series .= &#34;LINE2:&#39;bmem_realused&#39;#${conf[&#39;mem_available_color&#39;]}:&#39;Avail${rmspace}&#39; &#34;;
+
+        if ( $conf[&#39;graphreport_stats&#39;] ) {
+            $series .= &#34;CDEF:available_pos=bmem_available,0,INF,LIMIT &#34;
+                    . &#34;VDEF:available_last=available_pos,LAST &#34;
+                    . &#34;VDEF:available_min=available_pos,MINIMUM &#34; 
+                    . &#34;VDEF:available_avg=available_pos,AVERAGE &#34; 
+                    . &#34;VDEF:available_max=available_pos,MAXIMUM &#34; 
+                    . &#34;GPRINT:&#39;available_last&#39;:&#39; ${space1}Now\:%6.1lf%s&#39; &#34;
+                    . &#34;GPRINT:&#39;available_min&#39;:&#39;${space1}Min\:%6.1lf%s${eol1}&#39; &#34;
+                    . &#34;GPRINT:&#39;available_avg&#39;:&#39;${space2}Avg\:%6.1lf%s&#39; &#34;
+                    . &#34;GPRINT:&#39;available_max&#39;:&#39;${space1}Max\:%6.1lf%s\\l&#39; &#34;;
+        }
+    }
+
     $series .= &#34;LINE2:&#39;bmem_total&#39;#${conf[&#39;cpu_num_color&#39;]}:&#39;Total${rmspace}&#39; &#34;;
 
     if ( $conf[&#39;graphreport_stats&#39;] ) {
diff --git a/header.php b/header.php
index 91af0a2..729fe27 100644
--- a/header.php
+++ b/header.php
@@ -263,7 +263,7 @@ if ($initgrid or $gridwalk)
       # Use cookie so we dont have to pass gridstack around within this site.
       # Cookie values are automatically urlencoded. Expires in a day.
       if ( !isset($_COOKIE[&#34;gs&#34;]) or $_COOKIE[&#34;gs&#34;] != $gridstack_str )
-            setcookie(&#34;gs&#34;, $gridstack_str, time() + 86400);
+            setcookie(&#34;gs&#34;, $gridstack_str, time() + 86400, NULL, NULL, true, true);
    }
 
 # Invariant: back pointer is second-to-last element of gridstack. Grid stack
diff --git a/lib/GangliaAuth.php b/lib/GangliaAuth.php
index 4a4d1da..01ca10d 100644
--- a/lib/GangliaAuth.php
+++ b/lib/GangliaAuth.php
@@ -82,14 +82,14 @@ class GangliaAuth {
 
   // this is how a user &#39;logs in&#39;.
   public function setAuthCookie($user, $group=null) {
-    setcookie(&#39;ganglia_auth&#39;, json_encode( array(&#39;user&#39;=&gt;$user, &#39;group&#39;=&gt;$group, &#39;token&#39;=&gt;$this-&gt;getAuthToken($user)) ) );
+    setcookie(&#39;ganglia_auth&#39;, json_encode( array(&#39;user&#39;=&gt;$user, &#39;group&#39;=&gt;$group, &#39;token&#39;=&gt;$this-&gt;getAuthToken($user)) ), NULL, NULL, NULL, true, true );
     $this-&gt;user = $user;
     $this-&gt;group = $group;
     $this-&gt;tokenIsValid = true;
   }
 
   public function destroyAuthCookie() {
-    setcookie(&#39;ganglia_auth&#39;, &#39;&#39;, time());
+    setcookie(&#39;ganglia_auth&#39;, &#39;&#39;, time(), NULL, NULL, true, true);
     self::$auth = null;
   }
 
diff --git a/templates/default/cluster_host_metric_graphs.tpl b/templates/default/cluster_host_metric_graphs.tpl
index dbd80ab..95bc8aa 100644
--- a/templates/default/cluster_host_metric_graphs.tpl
+++ b/templates/default/cluster_host_metric_graphs.tpl
@@ -42,7 +42,7 @@
       {/foreach}
     {/if}
   {/if}
-  {if $index % $hostcols == 0}
+  {if $index % max(1, $hostcols) == 0}
     &lt;/tr&gt;&lt;tr&gt;
   {/if}
   {math &#34;$index + 1&#34; assign=&#34;index&#34;}
</code></pre>
  </div>
 </div> <!-- end .card-->

</div>
</div>
</div>
    </div>

        <div class="footer pt-4 text-white">
        <div class="container">
            <div class="d-flex align-items-center">
                <div>
                    <div>Powered by <a href="https://pagure.io/pagure" class="notblue">Pagure</a> 5.13.3</div>
                    <div>
                        <a href="https://docs.pagure.org/pagure/usage/index.html" class="notblue">Documentation</a> &bull;
                        <a href="https://pagure.io/pagure/new_issue" class="notblue">File an Issue</a> &bull;
                        <a href="/about">About this Instance</a> &bull;
                        <a href="/ssh_info" class="notblue">SSH Hostkey/Fingerprint</a>
                    </div>
                </div>
                <div class="ml-auto text-right">
                    <div>&copy; Red Hat, Inc. and others.</div>
                </div>
            </div>
        </div>
    </div>


    <script type="text/javascript" nonce="x8dmwigkFAnlN60QkSaKTGdHp" src="/static/vendor/jquery/jquery.min.js?version=5.13.3"></script>

    <script src="/static/vendor/bootstrap/bootstrap.bundle.min.js?version=5.13.3"></script>

    <script type="text/javascript" nonce="x8dmwigkFAnlN60QkSaKTGdHp">
      $('[data-toggle="tooltip"]').tooltip({placement : 'bottom'});
      $(".cancel_btn").click(function() {
        history.back();
      });
    </script>

<script type="text/javascript"  nonce="x8dmwigkFAnlN60QkSaKTGdHp" src="/static/vendor/lazyload/lazyload.min.js?version=5.13.3"></script>

<script type="text/javascript" nonce="x8dmwigkFAnlN60QkSaKTGdHp">
window.addEventListener("load", function(event) {
    lazyload();
});
</script>

<script type="text/javascript" nonce="x8dmwigkFAnlN60QkSaKTGdHp">
$("#giturl-toggle").on('click', function(event){
  event.stopPropagation();
  $("#giturl-more").toggle();
  $("#giturl-toggle").hide();
})

$(".fork_project_btn").click(function() {
  $('#fork_project').submit();
});

$(".select-on-focus").on("focus", function() {
  $(this).select();
});

</script>

<script type="text/javascript" nonce="x8dmwigkFAnlN60QkSaKTGdHp" src="/static/vendor/highlight.js/highlight.pack.js?version=5.13.3"></script>
<script type="text/javascript" nonce="x8dmwigkFAnlN60QkSaKTGdHp" src="/static/vendor/highlightjs-line-numbers/highlightjs-line-numbers.min.js?version=5.13.3"></script>
<script type="text/javascript" nonce="x8dmwigkFAnlN60QkSaKTGdHp" src="/static/vendor/highlight.js/spec.js?version=5.13.3"></script>

<script type="text/javascript" nonce="x8dmwigkFAnlN60QkSaKTGdHp">
  $(document).ready(function() {
  $('.fork_project_btn').click($("[name=fork_project]").submit);

  $('pre.syntaxhighlightblock code').each(function(i, block) {
    hljs.highlightBlock(block);
    hljs.lineNumbersBlock(block);
  });

  var cls = "highlighted-line";
  var lines = location.hash.substr(2).split('-').map(function (x) { return parseInt(x, 10) });
  if (! isNaN(lines[0]))
  {
    for (var i = lines[lines.length - 1]; i >= lines[0]; i--) {
      $('#_' + i).parent().parent().addClass(cls);
    }
    setTimeout(function(){
      $("#_" + lines[0]).get(0).scrollIntoView({behavior: "instant", block: "start", inline: "nearest"});
    }, 50);
  }
});
</script>

<script type="text/javascript" nonce="x8dmwigkFAnlN60QkSaKTGdHp">

  function updateHighlight() {
    var cls = "highlighted-line";
    $('.' + cls).removeClass(cls)
    if (location.hash !== '') {
      var lines = location.hash.substr(2).split('-').map(function (x) { return parseInt(x, 10) });
      for (var i = lines[lines.length - 1]; i >= lines[0]; i--) {
        $('[data-line-number=' + i + ']').closest('tr').addClass(cls);
      }
      return lines;
    }
    return [];
  }
  $(window).on('hashchange', updateHighlight);
  var selected = [];
  $("[data-line-number]").click(function (ev) {
    var line = $(this).attr('data-line-number');
    if (ev.shiftKey) {
      selected = selected.slice(-1).concat(line);
    } else {
      selected = [line];
    }

    var hash = '_' + selected[0];
    if (selected.length === 2) {
      hash = '_' + Math.min(selected[0], selected[1]) + '-' + Math.max(selected[0], selected[1]);
    }
    window.location.hash = hash;
    return false;
  });

</script>


</body>
</html>